src/main/java/com/patchworkmc/transformer/PatchworkTransformer.java

```diff
@@ -145,46 +143,19 @@ public void accept(String name, byte[] content) {
 			outputConsumer.accept(shimName, shimWriter.toByteArray());
 		});
 
-		HashMap<String, SubscribeEvent> subscribeEventStaticShims = new HashMap<>();
-		HashMap<String, SubscribeEvent> subscribeEventInstanceShims = new HashMap<>();
-
-		subscribeEvents.forEach(entry -> {
-			ClassWriter shimWriter = new ClassWriter(0);
-			String shimName = SubscribeEventGenerator.generate(name, entry, shimWriter);
-
-			if (subscribeEventStaticShims.containsKey(shimName) || subscribeEventInstanceShims.containsKey(shimName)) {
-				throw new UnsupportedOperationException("FIXME: Two @SubscribeEvent shims have the same name! This should be handled by Patchwork, it's a bug! Shim name: " + shimName);
-			}
+		boolean addedStaticEvent = false;
 
+		for (SubscribeEvent entry : subscribeEvents) {
```

Comment:
Why not just move the body of this loop to the lambda used in constructing the `EventHandlerRewriter`?